create_shards.c

```diff
@@ -84,17 +92,53 @@ master_create_distributed_table(PG_FUNCTION_ARGS)
 								  "foreign tables.")));
 	}
 
-	/* verify column exists in given table */
-	partitionColumnId = get_attnum(distributedTableId, partitionColumnName);
-	if (partitionColumnId == InvalidAttrNumber)
+	/* ColumnNameToColumn verifies column exists in given table */
+	partitionColumnVar = ColumnNameToColumn(distributedTableId, partitionColumnName);
+
+	if  (partitionMethod != HASH_PARTITION_TYPE)
 	{
-		ereport(ERROR, (errmsg("could not find column: %s", partitionColumnName)));
+		/* we only support hash partitioning right now */
+		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				        errmsg("pg_shard only supports hash partitioning")));
 	}
 
-	/* we only support hash partitioning method for now */
-	if (partitionMethod != HASH_PARTITION_TYPE)
+	/* depending on the partition type, check for the existence of support function */
+	if (partitionMethod == HASH_PARTITION_TYPE)
 	{
-		ereport(ERROR, (errmsg("unsupported partition method: %c", partitionMethod)));
+		Oid hashSupportFunction = SupportFunctionForColumn(partitionColumnVar,
+														   HASH_AM_OID, HASHPROC);
+		if (hashSupportFunction == InvalidOid)
+		{
+			Oid partitionColumnTypeId = partitionColumnVar->vartype;
```

Comment:
I don't see the point in this variable existing, so I'm going to remove it before merging (it's only referenced once).